src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int MULTIPLIER = 2;
+    private static final int INITIAL_CAPACITY = 16;
+    private int currentCapacity = INITIAL_CAPACITY;
+    private int size;
+    private Node<K, V> [] elements;
+
+    public MyHashMap() {
+        elements = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= currentCapacity * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> currentNode = elements[index];
+        while (currentNode != null) {
+            if ((key == null && currentNode.key == null)
+                    || (key != null && key.equals(currentNode.key))) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value, elements[index]);
```

Comment:
The usage of 'new Node<>(...)' is correct here as it maintains type inference. However, make sure that you understand why 'new Node<K,V>(...)' or 'new Node(...)' would be different and when each should be used.